homeassistant/components/climate/daikin.py

```diff
@@ -0,0 +1,275 @@
+    def name(self):
+        """Return the name of the thermostat, if any."""
+        return self._name
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement which this thermostat uses."""
+        return TEMP_CELSIUS
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        self._current_temperature = self.settings('htemp', True)
+
+        return self._current_temperature
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        self._target_temperature = self.settings('stemp', True)
+
+        return self._target_temperature
+
+    @property
+    def target_temperature_step(self):
+        """Return the supported step of target temperature."""
+        return 1
+
+    def set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        settings = {}
+
+        if kwargs.get(ATTR_OPERATION_MODE) is not None:
+            operation_mode = kwargs.get(ATTR_OPERATION_MODE)
+            current_operation = HA_STATE_TO_DAIKIN.get(operation_mode)
+            if current_operation is not None:
+                settings['mode'] = current_operation
+
+        if kwargs.get(ATTR_TEMPERATURE) is not None:
+            temperature = kwargs.get(ATTR_TEMPERATURE)
+            try:
+                self._target_temperature = int(temperature)
+                settings['stemp'] = str(self._target_temperature)
+            except ValueError:
+                _LOGGER.error("Invalid temperature %s", temperature)
+
+        if settings:
+            self._device.set(settings)
+            self.schedule_update_ha_state()
+
+    @property
+    def current_operation(self):
+        """Return current operation ie. heat, cool, idle."""
+        operation_mode = self._device.represent('mode')[1]
+        current_operation = DAIKIN_STATE_TO_HA.get(operation_mode)
+
+        return current_operation
+
+    @property
+    def operation_list(self):
+        """Return the list of available operation modes."""
+        return self._operation_list
+
+    def set_operation_mode(self, operation_mode):
+        """Set HVAC mode (auto, auxHeatOnly, cool, heat, off)."""
+        current_operation = HA_STATE_TO_DAIKIN.get(operation_mode)
+
+        if current_operation is not None:
+            self._device.set({"mode": current_operation})
+            self.schedule_update_ha_state()
+        else:
+            _LOGGER.error("Invalid operation mode %s", operation_mode)
+
+    @property
+    def current_humidity(self):
+        """Return the current humidity."""
+        self._current_humidity = self._device.represent('shum')[1]
+
+        return self._current_humidity
+
+    def set_humidity(self, humidity):
+        """Set new target temperature."""
+        if humidity is not None:
+            try:
+                self._target_humidity = int(humidity)
+                self._device.set({"shum": str(humidity)})
```

Comment:
You seem to be setting the same value as you return for `current_humidity`. However, `current_humidity` should return the actual humidity of the air, not the target humidity. Is this intended?